### ISSUE TITLE
Cancel out layer switch key when multiple keystrokes are on the same layer

### ISSUE THREAD

**Issue by sandsq** [sandsquaretech/AdvancedLayoutCalculator.jl#90]
```
L2------------------------------------
   ESC     Z     J     X   ENT   GRV
   TAB     K     F     P     S     C
  LCTL     L     O    LS     R     U
  SCLN     B     N     Q     M  COMM
    __    __   GUI   ALT    LS     Y
```
Suppose the above is layer 2. We can see that in a sequence like `for`, we would be hitting LS in between each letter, when in reality we would be holding down LS and then hitting `for`. This results in some technically inaccurate calculations -- for example, in `for`, `f` and `o` would be typed with the same finger but the score function does not detect that since it thinks there is an `LS` inbetween.

So basically if we have sequence of keys in the same layer, we should cancel out the LS that happens inbetween them and only count the initial one.

**Comment by sandsq** [comment]
But don't cancel if the layer switch is the same finger (apply same finger penalty)
